services/skus/service.go

```diff
@@ -676,10 +676,26 @@ func (s *Service) updateOrderStripeSession(ctx context.Context, dbi sqlx.ExtCont
 	return s.renewOrderStripe(ctx, dbi, ord, sub.ID, expt, paidt)
 }
 
-// CancelOrder cancels an order, propagates to stripe if needed.
-//
-// TODO(pavelb): Refactor this.
-func (s *Service) CancelOrder(orderID uuid.UUID) error {
+func (s *Service) CancelOrder(ctx context.Context, id uuid.UUID) error {
+	tx, err := s.Datastore.RawDB().BeginTxx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = tx.Rollback() }()
+
+	if err := s.cancelOrderTx(ctx, tx, id); err != nil {
```

Comment:
We're using a transaction here because a transaction is used implicitly anyway whether we do open it explicitly or not. It would happen even in a case of `SELECT`.

Arguably, changing a record is a unit of work, and since it's happening in a concurrent environment, being explicit about this makes even more sense.

Reply:
I think that is the entire point to the question, wrapping it in a transaction does absolutely nothing, its just extra code for no reason. Could you explain how executing a single sql statement is a unit if work?